reaper.go

```diff
@@ -416,17 +430,19 @@ func (r *reaper) affectedContainers(since time.Time, args filters.Args) ([]strin
 		if changed {
 			// Its not safe to remove a container which was created after
 			// the prune was initiated, as this may lead to unexpected behaviour.
-			return nil, fmt.Errorf("container %s: %w", container.ID, errChangesDetected)
+			errChanges = append(errChanges, fmt.Errorf("container %s: %w", container.ID, errChangesDetected))
+			continue
 		}
 
 		containerIDs = append(containerIDs, container.ID)
 	}
 
-	return containerIDs, nil
+	return containerIDs, errors.Join(errChanges...)
```

Comment:
IIUC, we are returning the error chain here, which means that all resources created after the prune init time will contribute an error here. And that error chain will contribute to the one in the `resources` method, but because they will be `errChangesDetected`, the pruneWait function will continue processing them as a log.Warn (see reaper.go:L329.

If this is assumption correct, then it's fine not returning a nil error but the error chain. Can you confirm that?

Reply:
There's a few impacts from this:
1. The returned won't be logged by the resources method
2. The error will contribute to the error chain and passed up
3. `puneWait` will retry if `errChangesDetected` is detected and shutdown timeout hasn't been reached

Reply:
Off back off this question I added code to ensure that shutdown timeout is enforced, triggering best effort clean up.

Let me know this makes sense?